include/PruManager.h

```diff
@@ -35,15 +41,50 @@ class PruManager
 };
 
 #if ENABLE_PRU_RPROC == 1
+/**
+ * \brief Support for interaction with PRU via
+ * Remote Proc + Mmap
+ * \nosubgrouping
+ */
 class PruManagerRprocMmap : public PruManager
 {
-// use rproc for start/stop and mmap for memory sharing
 public:
+	/**
+	 * Based on the value of pruNum chooses:<br>
+	 * 0 for PRUSS1 core 0<br>
+	 * 1 for PRUSS1 core 1<br>
+	 * 2 for PRUSS2 core 0<br>
+	 * 3 for PRUSS2 core 1<br>
+	 * and sets the appropriate values for the corresponding path variables
+	 */
 	PruManagerRprocMmap(unsigned int pruNum, int v);
+	/**
+	 * performs echo stop > state
+	 * and effectively stops the PRU
+	 */
 	void stop();
+	/**
+	 * checks whether to use McASP IRQ or non McASP IRQ PRU code and then calls
+	 * the next start function
+	 * @param useMcaspIrq Flag which<br>
+	 * if set `true` then passes McASP PRU IRQ firmware file path<br>
+	 * and if `false` then passes the non McASP IRQ PRU firmware path<br>
+	 * to the next start() function
+	 */
 	int start(bool useMcaspIrq);
+	/**
+	 * Load the firmware and start the PRU
+	 * @param path path to the appropriate PRU firmware `.out` file
```

Comment:
when moving this description to the base class, keep it generic "path to a PRU firmware file that is suitable to be loaded by the child class's underlying driver".

Reply:
then in the child class you can specify what the appropriate format is (i.e.: a binary of the firmware for `Uio` and an ELF `.out` suitable for rproc for the other one)

Reply:
Okay then I am leaving the base class' pruNum parameter empty as it is child class dependent.

Reply:
okay sure, done